processor/src/main/java/org/mapstruct/ap/model/source/IterableMapping.java

```diff
@@ -52,12 +53,7 @@ public static IterableMapping fromPrism(IterableMappingPrism iterableMapping, Ex
         if ( !elementTargetTypeIsDefined
             && iterableMapping.dateFormat().isEmpty()
             && iterableMapping.qualifiedBy().isEmpty() ) {
-            messager.printMessage(
-                Diagnostic.Kind.ERROR,
-                "'dateformat', 'qualifiedBy' and 'elementTargetType' are undefined in @IterableMapping, "
-                    + "define at least one of them.",
-                method
-            );
+            messager.printMessage( Diagnostic.Kind.ERROR, method, Message.iterablemapping_noelements );
```

Comment:
Now with the enum approach in place, couldn't also the severity/kind be part of the enum definition? I don't think one message is reported with different kinds at different places, so we can move it there and simplify the calls to printMessage().